front/src/utils/hooks/useLogo.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react';
+
+import defaultLogo from 'assets/logo-color.svg';
+import defaultOsrdLogo from 'assets/logo-osrd-color-white.svg';
+import proudLogo from 'assets/proud-logo-color.svg';
+import proudOsrdLogo from 'assets/proud-logo-osrd-color-white.svg';
+import xmasLogo from 'assets/xmas-logo-color.svg';
+import xmasOsrdLogo from 'assets/xmas-logo-osrd-color-white.svg';
+
+const MONTH_VALUES = {
+  JUNE: 5,
+  DECEMBER: 11,
+};
+
+const useLogo = () => {
+  const [digitalTwin, setDigitalTwin] = useState<{ logo: string; logoWithName: string }>({
+    logo: defaultLogo,
+    logoWithName: defaultOsrdLogo,
+  });
+  const [stdcm, setStdcm] = useState<{ stdcmLogo?: string; stdcmPngLogo?: string }>({
+    stdcmLogo: undefined,
+    stdcmPngLogo: undefined,
+  });
+  const [customizedDeployment, setCustomizedDeployment] = useState(false);
+
+  useEffect(() => {
+    const fetchInternalProd = async () => {
+      try {
+        const response = await fetch('/overrides/overrides.json');
+        if (!response.ok || response.headers.get('Content-Type') !== 'application/json') {
+          setCustomizedDeployment(false);
+          if (new Date().getMonth() === MONTH_VALUES.JUNE) {
+            setDigitalTwin({ logo: proudLogo, logoWithName: proudOsrdLogo });
+          }
+          if (new Date().getMonth() === MONTH_VALUES.DECEMBER) {
+            setDigitalTwin({ logo: xmasLogo, logoWithName: xmasOsrdLogo });
+          }
+        } else {
+          const overridesData = await response.json();
+          const { icons } = overridesData;
+
+          const lmrLogoPath = `/overrides/${icons.stdcm.light}.svg`;
+          const lmrPngLogoPath = `/overrides/${icons.stdcm.light}@2x.png`;
+          const horizonFullLogoPath = `/overrides/${icons.digital_twin.dark}.svg`;
+          const horizonLogoPath = `/overrides/${icons.digital_twin.dark}_logo.svg`;
+
+          setCustomizedDeployment(true);
+          setStdcm({ stdcmLogo: lmrLogoPath, stdcmPngLogo: lmrPngLogoPath });
+          setDigitalTwin({ logo: horizonLogoPath, logoWithName: horizonFullLogoPath });
+        }
+      } catch {
+        setCustomizedDeployment(false);
+      }
+    };
+    fetchInternalProd();
+  }, []);
+
+  return { digitalTwin, stdcm, customizedDeployment };
```

Comment:
It would be better to return a plain object like this:
```suggestion
  return { digitalTwinLogo, digitalTwinLogoWithName, stdcmLogo, stdcmSimulationSheetLogo, customizedDeployment };
```

Reply:
Done